tests/cpp_integration_tests/test_main_model.cpp

```diff
@@ -1373,67 +1374,117 @@ TEST_CASE("Test main model - runtime dispatch") {
-                CHECK(node_output_from_columnar[idx].u_pu == doctest::Approx(node_output_from_row_based[idx].u_pu));
-                // check update actually changed something
-                CHECK(node_output_from_columnar[idx].id == doctest::Approx(node_output_from_base[idx].id));
-                if (idx == 0) { // sym_load node
-                    CHECK(node_output_from_columnar[idx].u_pu == doctest::Approx(node_output_from_base[idx].u_pu));
-                } else {
-                    CHECK(node_output_from_columnar[idx].u_pu != doctest::Approx(node_output_from_base[idx].u_pu));
+            SUBCASE("Without IDs") {
+                ConstDataset update_data_with_ids{false, 1, "update", meta_data::meta_data_gen::meta_data};
+                update_data_with_ids.add_buffer("sym_load", sym_load_p_specified.size(), sym_load_p_specified.size(),
+                                                nullptr, nullptr);
+                update_data_with_ids.add_attribute_buffer("sym_load", "id", sym_load_ids.data());
+                update_data_with_ids.add_attribute_buffer("sym_load", "p_specified", sym_load_p_specified.data());
+
+                ConstDataset update_data_without_ids{false, 1, "update", meta_data::meta_data_gen::meta_data};
+                update_data_without_ids.add_buffer("sym_load", sym_load_p_specified.size(), sym_load_p_specified.size(),
+                                                   nullptr, nullptr);
+                update_data_without_ids.add_attribute_buffer("sym_load", "p_specified", sym_load_p_specified.data());
+
+                MainModel base_model{50.0, input_data};
+                MainModel columnar_model_w_id{base_model};
+                MainModel columnar_model_wo_id{base_model};
+
+                columnar_model_w_id.update_component<permanent_update_t>(update_data_with_ids);
+                columnar_model_wo_id.update_component<permanent_update_t>(update_data_without_ids);
+
+                std::vector<SymNodeOutput> node_output_columnar_w_id(state.node_input.size());
+                std::vector<SymNodeOutput> node_output_columnar_wo_id(state.node_input.size());
+
+                MutableDataset sym_output_columnar_w_id{true, 1, "sym_output", meta_data::meta_data_gen::meta_data};
+                sym_output_columnar_w_id.add_buffer("node", node_output_columnar_w_id.size(),
+                                                    node_output_columnar_w_id.size(), nullptr,
+                                                    node_output_columnar_w_id.data());
+                MutableDataset sym_output_columnar_wo_id{true, 1, "sym_output", meta_data::meta_data_gen::meta_data};
+                sym_output_columnar_wo_id.add_buffer("node", node_output_columnar_wo_id.size(),
+                                                     node_output_columnar_wo_id.size(), nullptr,
+                                                     node_output_columnar_wo_id.data());
+
```

Comment:
please also add a subcase where the output data contains no IDs column.

Reply:
More tests will be added and it will become more complete in the Row PR

Reply:
I am not sure that was the intended behavior.

Reply:
> I am not sure that was the intended behavior.

it was, because, for output data, all components are always output and even in the exact same order as for input data. the ids are even more redundant than for update data. users that care about memory so much that they'll remove it from update data will care even more about the output data

Reply:
row-based PR is OK

Reply:
Will be added in the Row PR.